the-collab-lab/tcl-68-smart-shopping-list
# List and Manage List styling

## Description

- Styled individual list items, add item form, list name header, share list form, delete list button, and List view conditional components (e.g., messages when list is empty or not selected)

## Related Issue

Closes #58 
Closes #59 

## Acceptance Criteria

The following elements should be styled on our `List` page according to our prototype:
- [x]  `Header`: List name
- [x]  `Label`: Search for an item (rename to Filter?)
- [x] `Input Search Box`: Search for an item by name
- [x] `Button`: Reset
- [x] `Label`: Add new item
- [x] `Input Text Box`: Item Name
- [x] `Input Select Dropdown`: Next Purchase Date
- [x] `Button`: Add item
- [x] `List Items`: Items in the user's list
> - [x] Checkbox
> - [x] Icon
> - [x] Item Name
> - [x] Purchase Urgency
> - [x] `Button`: Delete
- [x] `Button`: Select a list
- [x] `Header`: List is currently empty
- [x] `Header`: You haven't selected a list

The following elements from our `Manage List` page should be styled according to our prototype:

- [x] `Header`: List name
~~- [ ] `Header`: Manage List~~ _(We opted to exclude this design element.)_
- [x] `Label`: Share List
- [x] `Input Text Box`: Enter Recipient's Email Address
- [x] `Button`: Send Invite!
- [x] `Button`: Delete List

## Type of Changes

`enhancement`

## Updates

### Before

List view - no list selected
![image](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/98671035/131c54c7-74c0-454f-a5f1-8c9be07b1de2)

Manage List view - no list selected
![image](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/98671035/cda0608d-b044-49d5-bbf8-b910b16fb362)

List view - empty list
![image](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/98671035/1c5aa6f2-fed5-4ee2-adbe-cccf1aaf8fd6)

List view - populated list
![image](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/98671035/89fedf18-aba6-4655-a543-5a47e8e16c72)

Managed List view - list selected
![image](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/98671035/bb857c40-ee80-4e5a-ae05-fde4cfe2f549)

### After
List view - no list selected
![image](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/98671035/db2788e5-8659-424b-bb26-4ba74775db1f)

Manage List view - no list selected (should look the same as List view)
![image](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/98671035/cfef4819-0729-456f-8f83-9cc25678beaf)

List view - empty list
![image](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/98671035/d1be7430-b837-47a4-b72d-dc92d96d38f8)

List view - populated list
![image](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/98671035/485b0076-d221-4233-885b-27afabd9607d)

Manage List view - list selected
![image](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/98671035/d3508e13-e1c5-4935-b200-6d24656bc486)

## Testing Steps / QA Criteria

- Open deployment link or run branch `ap-et-styling` locally
- Ensure local storage is cleared
- Without clicking on a list name, navigate to both the List and Manage List pages and verify that they look like the screenshots
    - Verify that the 'select a list' button still takes user to the Home page
- Click on a list you know is empty, navigate to the List page and verify that it looks like the screenshot
- Click on a list with items, navigate to the List and Manage List pages and verify that they look like the screenshots
- Verify that all the functionality (e.g. adding an item, purchasing an item, deleting an item, sharing a list, deleting a list) still works as expected
